### PR TITLE
Document ability to explicitly declare Deconstruct method.

### PR DESCRIPTION
**Pull request by AlekseyTs**

### CODE REVIEW

File: proposals/records.md [hunk 1]
```diff
@@ -238,11 +238,13 @@ For a record:
...
 ### Deconstruct
 
-A positional record synthesizes a public void-returning method called Deconstruct with an out
+A positional record synthesizes a public void-returning instance method called Deconstruct with an out
 parameter declaration for each parameter of the primary constructor declaration. Each parameter
 of the Deconstruct method has the same type as the corresponding parameter of the primary
 constructor declaration. The body of the method assigns each parameter of the Deconstruct method
 to the value from an instance member access to a member of the same name.
+The method can be declared explicitly. It is an error if the explicit declaration does not match
+the expected signature or accessibility, or is static.
```
Comment (RikkiGibson) [inline proposals/records.md:247]: Do we still permit the user to write a Deconstruct method with a different number of parameters or different type of parameters?

Comment (333fred) [reply proposals/records.md:247]: Good question. I would expect to be able to declare my own `Deconstruct` that adds a nominal property, for example.

Comment (AlekseyTs) [reply proposals/records.md:247]: > Do we still permit the user to write a Deconstruct method with a different number of parameters or different type of parameters?

Yes.

**Comment by AlekseyTs** [comment]
@MadsTorgersen, @cston, @jcouv Please review

File: proposals/records.md (hunk 1 above)
Comment (333fred) [reply proposals/records.md:247]: What would be the error case for incorrect signature then? A conflicting accessibility/staticness would produce an error, but a different parameter type or number of parameters should produce no error, right?

Comment (AlekseyTs) [reply proposals/records.md:247]: > What would be the error case for incorrect signature then? 

Wrong error type, for example.

> A conflicting accessibility/staticness would produce an error, but a different parameter type or number of parameters should produce no error, right?

Correct.

Comment (333fred) [reply proposals/records.md:247]: Thanks for clarifying.

**Pull request merged**